modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ModalAddObjectField.tsx

```diff
@@ -0,0 +1,260 @@
+		if (
+			field.businessType === 'Aggregation' ||
+			field.businessType === 'Formula'
+		) {
+			field.readOnly = 'true';
+			delete field.readOnlyConditionExpression;
+		}
+
+		if (!Liferay.FeatureFlags['LPS-170122']) {
+			delete field.readOnly;
+			delete field.readOnlyConditionExpression;
+		}
+
+		if (field.label) {
+			field = {
+				...field,
+				name:
+					field.name ||
+					toCamelCase(field.label[defaultLanguageId] as string, true),
+			};
+
+			delete field.listTypeDefinitionId;
+
+			try {
+				await API.save(apiURL, field, 'POST');
+
+				onClose();
+				window.location.reload();
+			}
+			catch (error) {
+				setError((error as Error).message);
+			}
+		}
+	};
+
+	const {
+		errors,
+		handleChange,
+		handleSubmit,
+		setValues,
+		values,
+	} = useObjectFieldForm({
+		initialValues,
+		onSubmit,
+	});
+
+	const applyFeatureFlag = () => {
+		return objectFieldTypes.filter((objectFieldType) => {
+			if (!Liferay.FeatureFlags['LPS-164948']) {
+				return objectFieldType.businessType !== 'Formula';
+			}
+		});
+	};
+
+	const showEnableTranslationToggle =
+		values.businessType === 'LongText' ||
+		values.businessType === 'RichText' ||
+		values.businessType === 'Text';
+
+	useEffect(() => {
+		if (!objectDefinition) {
+			const makeFetch = async () => {
+				const objectDefinitionResponse = await API.getObjectDefinitionByExternalReferenceCode(
+					objectDefinitionExternalReferenceCode
+				);
+
+				setObjectDefinition(objectDefinitionResponse);
+			};
+
+			makeFetch();
+		}
+
+		if (Liferay.FeatureFlags['LPS-172017']) {
+			if (
+				objectDefinition?.enableLocalization &&
+				showEnableTranslationToggle
+			) {
+				setValues({
+					localized: true,
+				});
+
+				return;
+			}
+
+			setValues({
+				localized: false,
+			});
+
+			return;
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [values.businessType]);
+
+	return (
+		<ClayModalProvider>
+			<ClayModal center observer={observer}>
+				<ClayForm onSubmit={handleSubmit}>
+					<ClayModal.Header>
+						{Liferay.Language.get('new-field')}
+					</ClayModal.Header>
+
+					<ClayModal.Body>
+						{error && (
+							<ClayAlert displayType="danger">{error}</ClayAlert>
+						)}
+
+						<Input
+							error={errors.label}
+							label={Liferay.Language.get('label')}
+							name="label"
+							onChange={({target: {value}}) => {
+								setValues({
+									label: {[defaultLanguageId]: value},
+								});
+							}}
+							required
+							value={values.label?.[defaultLanguageId]}
+						/>
+
+						<ObjectFieldFormBase
+							errors={errors}
+							handleChange={handleChange}
+							objectDefinition={objectDefinition}
+							objectDefinitionExternalReferenceCode={
+								objectDefinitionExternalReferenceCode
+							}
+							objectField={values}
+							objectFieldTypes={
+								!Liferay.FeatureFlags['LPS-164948']
+									? applyFeatureFlag()
+									: objectFieldTypes
+							}
+							objectName={objectName ?? ''}
```

Comment:
@danvb112 You could also define a default value at line 35 like: `objectName = '',` but this is fine for me :)

Reply:
Thanks for the suggestion!